core/host_api/impl/child_storage_extension.cpp

```diff
@@ -234,4 +234,95 @@ namespace kagome::host_api {
     return memory.storeBuffer(root);
   }
 
+  void ChildStorageExtension::ext_default_child_storage_clear_prefix_version_1(
+      runtime::WasmSpan child_storage_key, runtime::WasmSpan prefix) {
+    auto &memory = memory_provider_->getCurrentMemory()->get();
+    auto [child_key_buffer, prefix_buffer] =
+        loadBuffer(memory, child_storage_key, prefix);
+
+    SL_TRACE_VOID_FUNC_CALL(logger_, child_key_buffer, prefix);
+
+    auto result = executeOnChildStorage<std::tuple<bool, uint32_t>>(
+        child_key_buffer,
+        [](auto &child_batch, auto &prefix) {
+          return child_batch->clearPrefix(prefix, std::nullopt);
+        },
+        prefix_buffer);
+
+    if (not result) {
+      logger_->error(
+          "ext_default_child_storage_clear_prefix_version_1 failed with "
+          "reason: {}",
+          result.error().message());
+    }
+  }
+
+  runtime::WasmSpan
+  ChildStorageExtension::ext_default_child_storage_read_version_1(
+      runtime::WasmSpan child_storage_key,
+      runtime::WasmSpan key,
+      runtime::WasmSpan value_out,
+      runtime::WasmOffset offset) const {
+    auto &memory = memory_provider_->getCurrentMemory()->get();
+    auto [child_key_buffer, key_buffer] =
+        loadBuffer(memory, child_storage_key, key);
+    auto [value_ptr, value_size] = runtime::PtrSize(value_out);
+
+    auto read = executeOnChildStorage<common::Buffer>(
+        child_key_buffer,
+        [](auto &child_batch, auto &key) { return child_batch->get(key); },
+        key_buffer);
+    std::optional<uint32_t> res{std::nullopt};
+    if (read) {
```

Comment:
If there's an error other than NO_VALUE, it would be silenced here, as far as I can see. I suggest you use tryGet instead and log occurring errors.

Reply:
But get() is just a wrapper for tryGet() that peels off optional. Even if I will use tryGet() here, I'll just write get() logic without changes. What's the point?